app/code/Magento/UrlRewriteGraphQl/Model/Resolver/UrlRewrite.php

```diff
@@ -74,17 +74,20 @@ public function resolve(
         };
         
         if (isset($args['url'])) {
-            $url = $args['url'];
-            if (substr($url, 0, 1) === '/' && $url !== '/') {
-                $url = ltrim($url, '/');
+            $parsedUrl = parse_url($args['url']);
+            $url = $parsedUrl['path'];
+            if ($parsedUrl['path'][0] === '/' && $parsedUrl['path'] !== '/') {
```

Comment:
`$url` variable can be used here as before, no need to change lines 78-79 (from the original revision).